.github/workflows/ci.yml

```
@@ -0,0 +1,29 @@
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+name: CI
+
+jobs:
+
+  build_and_test:
+    name: build and run tests
+    runs-on: ubuntu-latest
+    env:
+      DEBIAN_FRONTEND: noninteractive
+    steps:
+      - uses: actions/checkout@v1
+      - name: Install qt and build tools
+        run: |
+          sudo apt-get update &&
+          sudo apt-get install -y --no-install-recommends \
+            pkg-config g++ make \
+            qt5-default qttools5-dev-tools
+      - name: Run qmake
+        run: mkdir build && cd build && qmake -r ..
```

Make CI fail on warnings:

```suggestion
        run: mkdir build && cd build && qmake -r .. "QMAKE_CXXFLAGS=-Werror"
```